packages/ckeditor5-link/src/linkediting.js

```diff
@@ -414,4 +418,86 @@ export default class LinkEditing extends Plugin {
 			}
 		} );
 	}
+
+	/**
+	 * Starts listening to {@link module:engine/model/model~Model#deleteContent} and {@link module:engine/model/model~Model#insertContent}
+	 * and checks whether typing over the link. If so, attributes of removed text are preserved and applied to the inserted text.
+	 *
+	 * The purpose of this action is to allow modifying a text without loosing the `linkHref` attribute (and other).
+	 *
+	 * See https://github.com/ckeditor/ckeditor5/issues/4762.
+	 *
+	 * @private
+	 */
+	_enableTypingOverLink() {
+		const editor = this.editor;
+
+		// Selection attributes when started typing over the link.
+		let selectionAttributes;
+
+		// Whether pressed `Backspace` or `Delete`. If so, attributes should not be preserved.
+		let deletedContent;
+
+		// Detect pressing `Backspace` / `Delete`.
+		editor.editing.view.document.on( 'delete', ( evt, data ) => {
+			/* istanbul ignore else */
+			if ( data.domEvent.keyCode == keyCodes.delete || data.domEvent.keyCode == keyCodes.backspace ) {
+				deletedContent = true;
+			}
+		}, { priority: 'high' } );
+
+		editor.model.on( 'deleteContent', () => {
```

Comment:
Use `this.listenTo( editor.model, ... )` - we use this pattern elsewhere. The difference is that it allows to stop all `LinkEditing` listeners on other emitters by `this.stopListening()` - ie when  calling `Plugin.destroy()`.

There might be other problems of which I don't recall now.